client/my-sites/stats/features/modules/stats-module-devices/stats-module-devices.tsx

```diff
@@ -0,0 +1,329 @@
+		}
+
+		return {
+			id: device.key,
+			value: device.value,
+			icon,
+			name: device.label,
+			descrtipion: device.label,
+			className: `donut-${ device.key }`,
+		};
+	} );
+};
+
+const StatsModuleDevices: React.FC< StatsModuleDevicesProps > = ( {
+	path,
+	className,
+	isLoading,
+	query,
+} ) => {
+	const { devices: devicesStrings } = statsStrings();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const translate = useTranslate();
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
+	const isNewEmptyStateEnabled = config.isEnabled( 'stats/empty-module-traffic' );
+
+	const optionLabels = {
+		[ OPTION_KEYS.SIZE ]: {
+			selectLabel: translate( 'Size' ),
+			headerLabel: translate( 'Visitors share per screen size' ),
+			analyticsId: 'screensize',
+		},
+		[ OPTION_KEYS.BROWSER ]: {
+			selectLabel: translate( 'Browser' ),
+			headerLabel: translate( 'Browser' ),
+			analyticsId: 'browser',
+		},
+		[ OPTION_KEYS.OS ]: {
+			selectLabel: translate( 'OS' ),
+			headerLabel: translate( 'Operating System' ),
+			analyticsId: 'os',
+		},
+	};
+
+	const [ selectedOption, setSelectedOption ] = useState( OPTION_KEYS.SIZE );
+	const { isFetching, data } = useModuleDevicesQuery( siteId, selectedOption, query );
+	const showLoader = isLoading || isFetching;
+
+	const changeViewButton = ( selection: SelectOptionType ) => {
+		const filter = selection.value;
+
+		// publish an event
+		const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
+		recordTracksEvent( `${ event_from }_devices_module_menu_clicked`, {
+			button: optionLabels[ filter ]?.analyticsId ?? filter,
+		} );
+
+		setSelectedOption( filter );
+	};
+
+	const toggleControlComponent = (
+		<SimplifiedSegmentedControl
+			className="stats-module-devices__tabs"
+			options={ Object.entries( optionLabels ).map( ( entry ) => ( {
+				value: entry[ 0 ], // optionLabels key
+				label: entry[ 1 ].selectLabel, // optionLabels object value
+			} ) ) }
+			initialSelected={ selectedOption }
+			// @ts-expect-error TODO: missing TS type
+			onSelect={ changeViewButton }
+		/>
+	);
+
+	const chartData = OPTION_KEYS.SIZE === selectedOption ? prepareChartData( data ) : null;
+
+	return (
+		<>
+			{ isNewEmptyStateEnabled && (
+				<>
+					{ showLoader && (
+						<StatsCardSkeleton
+							isLoading={ isFetching }
+							className={ className }
+							title={ devicesStrings.title }
+							type={ 3 }
+						/>
+					) }
+					{ ! showLoader &&
```

Comment:
The code duplication from L167 to L258 is getting a bit out of control. Is there anything we could easily reuse with reasonable efforts here?

Reply:
I copied everything for two reasons: to make it easier to remove the future flag and to avoid the risk of leaking new changes. I believe that this is not only convenient but also safer.
Of course, we could minimise it, but future cleaning won't be as straightforward as removing a whole block of the old code.

I think we should keep it as it is here and focus on cleaning the future flags quickly, but if you feel strongly about it, I can work on it more to prioritise less code.

Reply:
I'm happy to ship and iterate 🙂

Just want to clarify that I was not talking about the branch for the feature flag. If you look at L167 to L258, two components using the same component `StatsCard`, which I suppose we created to unify the logic.

Reply:
Lines 167 to 191 contain a new empty state. Lines 192 to 258 have the old component.
I'll update this to avoid repetition.

Reply:
Done in 633205f7546ca1f4b3bfb2050e11896d2ec47bb0